### PR TITLE
Convert 3 enums to enum classes

### PR DESCRIPTION
**Pull request by anothersimulacrum**
#### Summary
SUMMARY: None

#### Purpose of change
Make way for #43144

#### Describe the solution
Convert enums to enum classes, so the compiler doesn't attempt to create innapropriate functions with them.

#### Testing
Game compiles without error.

### CODE REVIEW

**Comment by ZhilkinSerg** [comment]
We need to change `TRUE` damage identifier to something else - some build environments will choke on that name.

My suggestion for new name would be `PURE`.

File: src/damage.cpp [hunk 1]
```diff
@@ -25,19 +25,19 @@ std::string enum_to_string<damage_type>( damage_type data )
...
-        case damage_type::DT_BASH: return "bash";
-        case damage_type::DT_CUT: return "cut";
-        case damage_type::DT_ACID: return "acid";
-        case damage_type::DT_STAB: return "stab";
-        case damage_type::DT_HEAT: return "heat";
-        case damage_type::DT_COLD: return "cold";
-        case damage_type::DT_ELECTRIC: return "electric";
-        case damage_type::DT_BULLET: return "bullet";
+        case damage_type::NONE: return "none";
+        case damage_type::PURE: return "true";
```
Comment (ZhilkinSerg) [inline src/damage.cpp:29]: Will you update text to `pure` here and in JSON later?

File: src/damage.cpp [hunk 2]
```diff
@@ -259,24 +259,24 @@ float resistances::get_effective_resist( const damage_unit &du ) const
...
-    { translate_marker_context( "damage type", "biological" ), DT_BIOLOGICAL },
-    { translate_marker_context( "damage type", "bash" ), DT_BASH },
-    { translate_marker_context( "damage type", "cut" ), DT_CUT },
-    { translate_marker_context( "damage type", "acid" ), DT_ACID },
-    { translate_marker_context( "damage type", "stab" ), DT_STAB },
-    { translate_marker_context( "damage_type", "bullet" ), DT_BULLET },
-    { translate_marker_context( "damage type", "heat" ), DT_HEAT },
-    { translate_marker_context( "damage type", "cold" ), DT_COLD },
-    { translate_marker_context( "damage type", "electric" ), DT_ELECTRIC }
+    { translate_marker_context( "damage type", "true" ), damage_type::PURE },
```
Comment (ZhilkinSerg) [inline src/damage.cpp:270]:
```suggestion
    { translate_marker_context( "damage type", "pure" ), damage_type::PURE },
```

**Comment by jbytheway** [comment]
Can damage types appear in saves (e.g. for artifacts, maybe)?  If so, we need save migration if this name is going to change.

**Comment by KorGgenT** [comment]
artifacts won't have damage types since they indirect to spells at least, the worst they'd have are the enchantment values, which i don't see any changes to here.

**Comment by ZhilkinSerg** [comment]
There are serialize/deserialize functions for `damage_over_time_data`.

**Comment by anothersimulacrum** [comment]
Changed to pure, added migration.

**Pull request merged**